src/components/StructureExplorer/DrawStructure.tsx

```diff
@@ -41,23 +42,31 @@ function Search({ search, setSearch }: SearchProps): JSX.Element {
 
 function Board({ setIdCode }: BoardProps): JSX.Element {
   const BREAKPOINT = 1024;
+  const windowSize = useRef(window.innerWidth);
   const refParent = useRef<HTMLDivElement>(null);
 
-  const [boardWidth, setBoardWidth] = useState(
-    window.innerWidth <= BREAKPOINT ? refParent.current?.offsetWidth : 470,
+  const [boardWidth, setBoardWidth] = useState<number>(
+    (windowSize.current <= BREAKPOINT && refParent.current?.offsetWidth) || 470,
   );
 
+  const handleResize = (ref: ObservedSize) =>
+    setBoardWidth((windowSize.current <= BREAKPOINT && ref.width) || 470);
+
+  useResizeObserver<HTMLDivElement>({
+    ref: refParent,
+    onResize: (ref) => {
+      handleResize(ref);
+    },
+  });
+
   useEffect(() => {
-    const handleResize = () =>
+    if (refParent.current) {
       setBoardWidth(
-        window.innerWidth <= BREAKPOINT ? refParent.current?.offsetWidth : 470,
+        (window.innerWidth <= BREAKPOINT && refParent.current.offsetWidth) ||
+          470,
       );
-    window.addEventListener('resize', handleResize);
-    return () => {
-      window.removeEventListener('resize', handleResize);
-    };
+    }
   }, []);
```

Comment:
This is all overly complicated. You should be able to rely on the value returned by `useResizeObserver` and compute what you need from it (no additional useState, useEffect, or useRef)